components/pango_vars/include/pangolin/var/var.h

```diff
@@ -167,4 +168,95 @@ inline std::ostream& operator<<(std::ostream& s, Var<T>& rhs)
   return s;
 }
 
+template <int N>
+class VarVector
+{
+  public:
+  static_assert(N >= 1);
+  VarVector(
+      std::string name, Eigen::Vector<double, N> const& init,
+      Eigen::Vector<double, N> const& min, Eigen::Vector<double, N> const& max)
+  {
+    for (int i = 0; i < N; ++i) {
+      var_vec_[i] = std::make_optional(
+          Var<double>(FARM_FORMAT("{}[{}]", name, i), init[i], min[i], max[i]));
+    }
+  }
+
+  VarVector(
+      std::string name, Eigen::Vector<double, N> const& init, double min,
+      double max) :
+      VarVector(
+          name, init, (min * Eigen::Vector<double, N>::Ones()).eval(),
+          (max * Eigen::Vector<double, N>::Ones()).eval())
+  {
+  }
+
+  bool hasGuiChanged()
+  {
+    for (int i = 0; i < N; ++i) {
+      if (FARM_UNWRAP(var_vec_[i]).GuiChanged()) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  void set(Eigen::Vector<double, N> const& init)
+  {
+    for (int i = 0; i < N; ++i) {
+      auto& var = FARM_UNWRAP(var_vec_[i]);
+      var = init[i];
+    }
+  }
+
+  Eigen::Vector<double, N> get() const
+  {
+    Eigen::Vector<double, N> var;
+    for (int i = 0; i < N; ++i) {
+      var[i] = FARM_UNWRAP(var_vec_[i]);
+    }
+    return var;
+  }
+
+  private:
+  std::array<std::optional<Var<double>>, N> var_vec_;
+};
+
+class VarSe3
```

Comment:
I'd rather this were elsewhere to keep Var simple. I plan on introducing a traits system for stuff like this soon